XenAdmin/Diagnostics/Checks/DmcCheck.cs

```diff
@@ -0,0 +1,103 @@
+ *     following disclaimer in the documentation and/or other 
+ *     materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using XenAdmin.Core;
+using XenAdmin.Diagnostics.Hotfixing;
+using XenAdmin.Diagnostics.Problems;
+using XenAdmin.Diagnostics.Problems.HostProblem;
+using XenAdmin.Diagnostics.Problems.PoolProblem;
+using XenAPI;
+
+namespace XenAdmin.Diagnostics.Checks
+{
+    class DmcCheck : PoolCheck
+    {
+        private readonly Pool _pool;
+        private readonly Dictionary<string, string> _installMethodConfig;
+        private readonly bool _manualUpgrade;
+        private readonly Control _control;
+
+        public DmcCheck(Control control, Pool pool, Dictionary<string, string> installMethodConfig, bool manualUpgrade)
+            : base(pool)
+        {
+            _control = control;
+            _pool = pool;
+            _installMethodConfig = installMethodConfig;
+            _manualUpgrade = manualUpgrade;
+        }
+
+        public override bool CanRun()
+        {
+            return !Helpers.Post82X(_pool.Connection) && !Helpers.FeatureForbidden(_pool.Connection, Host.RestrictDMC);
+        }
+
+        protected override Problem RunCheck()
+        {
+            var coordinator = Helpers.GetCoordinator(Pool.Connection);
+
+            if (!_manualUpgrade)
+            {
+                var hotfix = HotfixFactory.Hotfix(coordinator);
+                if (hotfix != null && hotfix.ShouldBeAppliedTo(coordinator))
+                    return new HostDoesNotHaveHotfixWarning(this, coordinator);
+            }
+
+            var vms = _pool.Connection.Cache.VMs.Where(v =>
+                (v.power_state == vm_power_state.Running ||
+                 v.power_state == vm_power_state.Paused ||
+                 v.power_state == vm_power_state.Suspended) &&
+                (v.memory_static_min > v.memory_dynamic_min || //corner case, probably unlikely
+                 v.memory_dynamic_min != v.memory_dynamic_max ||
+                 v.memory_dynamic_max != v.memory_static_max)).ToList();
+
+            if (vms.Count == 0)

```

Comment:
Nit: I've seen recommendations (from Resharper or SonarQube I forget which) that this is better:
```csharp
if (!vms.Any())
```

Reply:
Any() is definitely preferable to linq's Count() method, so I expect a warning for this, but does it warn for the Count property too? The latter is more efficient as it does not recalculate.

Reply:
Fair point, yes the efficiency of linq's `Count()` was what I was concerned with.
I will mark the thread as resolved.